### PR TITLE
Add support for long text

### PR DESCRIPTION
**Pull request by Sukhendu2002**
Issue: https://github.com/PradyumnaKrishna/Enigma-Protocol/issues/20

### CODE REVIEW

File: client/src/views/Chat.vue [hunk 1]
```diff
@@ -289,7 +313,19 @@ export default {
...
-      const message = this.privateKey.decrypt(data.message);
+      const encryptedKey = data.key;
+
+      //decrypts the symmetric key using their private key.
+      const key = this.privateKey.decrypt(encryptedKey);
+      const iv = data.iv;
+      //decrypts the message using the symmetric key.
+      const decipher = forge.cipher.createDecipher("AES-CBC", key);
+      decipher.start({ iv: iv });
+      decipher.update(data.message);
```
Comment (PradyumnaKrishna) [inline client/src/views/Chat.vue:324]: This is the answer we are looking for.

https://github.com/digitalbazaar/forge/issues/315#issuecomment-152741165

```suggestion
      decipher.update(forge.util.createBuffer(data.message));
```

**Review by PradyumnaKrishna — CHANGES_REQUESTED** [review]
This looks solid @Sukhendu2002, well done!

Mentioned the possible fix to your problem.

Also, this encryption function looks big to me. Can we move this to a new library called `crypto.js` or `crypto.ts`? A function that returns encrypted `encrypted`, `iv` and `key` as a json, and vice-versa for decryption.

File: client/src/views/Chat.vue (hunk 1 above)
Comment (Sukhendu2002) [reply client/src/views/Chat.vue:324]: ok, thanks for the review, I will make the changes

**Comment by Sukhendu2002** [comment]
@PradyumnaKrishna, I make the changes.

So to solve this issue I generate a random symmetric key and encrypt the message with the symmetric key and then encrypt the symmetric key with the recipient public key.

For the receiver end, I am sending the symmetric key along with iv and the encrypted message, the receiver has to first decrypt the symmetric key using their private key and then decrypt the message using the decrypted symmetric key.

**Review by PradyumnaKrishna — APPROVED** [review]
Thanks @Sukhendu2002, This looks good to me.

LGTM

**Pull request merged**